### DISCUSSION TITLE
do the global pooling operators no longer work with a null batch?

### DISCUSSION THREAD

**Discussion by davidireland-iso** [pyg-team/pytorch_geometric#4157] [Q&A]
I used to use the code 
```
summary_1_mean = global_mean_pool(x, batch)
summary_1_max = global_max_pool(x, batch)
```
which would work if the batch was `None`, but in the latest version of PyG this gives me an error as a tensor is expected for the dimensions. I can hard code my way around this, but it was nice to not have to do this in the older version. 

The use case is that my network is designed for obtaining graph level embedding and so the batch was needed when training, but at inference I was often interested in obtaining an embedding of a single graph at a time so rather than set up a `DataLoader` object of just one graph I could make a `Data` object and set `_DataObj_.batch=None` and it would work nicely. 

Is there a reason this has been removed from the latest version?

**Comment by rusty1s** [comment]
Thanks for reporting. I'm not sure why and when it was removed TBH. Nonetheless, I agree this is an important feature to have, so I added it back in via https://github.com/pyg-team/pytorch_geometric/pull/4161.
